### PR TITLE
add support to extend context with xpos rope

### PR DESCRIPTION
**Pull request by winglian**
I'm able to generate a PoC with openllama extended to 4k with this (will release soon)

### CODE REVIEW

File: src/axolotl/utils/models.py [hunk 1]
```diff
@@ -127,6 +127,14 @@ def load_model(
         # TODO: Check if this would overwrite previous additional_special_tokens
         tokenizer.add_special_tokens({"additional_special_tokens": [MEM_TOKEN]})
 
+    if cfg.is_llama_derived_model and cfg.xpos_rope:
```
Comment (NanoCode012) [inline src/axolotl/utils/models.py:130]: Perhaps it should be `elif` from above. We should maybe not allow multiple patches?

Comment (winglian) [reply src/axolotl/utils/models.py:130]: it works with xformers_attention,it's specific to the rotary embedding layer which most of the above doesn't touch

Comment (NanoCode012) [reply src/axolotl/utils/models.py:130]: Oh ok!

Comment (NanoCode012) [reply src/axolotl/utils/models.py:130]: Could you also add this to Readme please?

**Pull request merged**